metrics/ci_integrations/lib/source/github_actions/adapter/github_actions_source_client_adapter.dart

```diff
@@ -252,15 +255,20 @@ class GithubActionsSourceClientAdapter
   }
 
   /// Maps the given [job] and [run] to the [BuildData] instance.
-  Future<BuildData> _mapJobToBuildData(
+  BuildData _mapJobToBuildData(
     WorkflowRunJob job,
     WorkflowRun run,
-  ) async {
+  ) {
+    final buildStatus = _mapJobToBuildStatus(job);
+    final duration = buildStatus == BuildStatus.inProgress
+        ? null
+        : _calculateJobDuration(job);
```

Comment:
NIT: would it be a bit more readable? 
```suggestion
    final duration = buildStatus != BuildStatus.inProgress
        ? _calculateJobDuration(job)
        : null;
```